src/queue.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+class Node {
+  constructor(data, follows, proceeds) {
+    this.follows = follows
+    this.data = data
+    this.proceeds = proceeds
+  }
+}
+
```

Comment:
The use of the Node class is a brilliant manner of separating the responsibility of the functions